spec/support/controller_spec_helper.rb

```diff
@@ -9,9 +9,9 @@ def pagination_pattern
   end
 
   def login_as_api_user(user = User.new)
-    request.env['warden'] = double('Authenticated API User',
-                                   authenticate!: true,
-                                   authenticated?: true,
-                                   user:)
+    double('Authenticated API User',
```

Comment:
I don't see the value of this `request.env['warden']`, so I've removed it and see what happens.